sdcm/utils/raft/common.py

```diff
@@ -29,3 +40,117 @@ def validate_raft_on_nodes(nodes: list["BaseNode"]) -> None:
+        if not self.bootstrap_node.stop_wait_db_up_event.is_set():
+            self.bootstrap_node.stop_wait_db_up_event.set()
+        LOGGER.debug("Stop event was set for node %s", self.bootstrap_node.name)
+
+    def _start_bootstrap(self, timeout=3600):
+        try:
+            LOGGER.debug("Starting bootstrap process %s", self.bootstrap_node.name)
+            self.bootstrap_node.parent_cluster.node_setup(self.bootstrap_node, verbose=True, timeout=timeout)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.error("Setup failed for node %s with err %s", self.bootstrap_node.name, exc)
+        finally:
+            self._set_wait_stop_event()
+        LOGGER.debug("Node %s was bootstrapped", self.bootstrap_node.name)
+
+    def _abort_bootstrap(self, abort_action: Callable, log_message: str, timeout: int = 600):
+        LOGGER.debug("Stop bootsrap process after log message: '%s'", log_message)
+        log_follower = self.bootstrap_node.follow_system_log(patterns=[log_message])
+        try:
+            wait_for(func=lambda: list(log_follower), step=5,
+                     text="Waiting log message to stop scylla...",
+                     timeout=timeout,
+                     throw_exc=True,
+                     stop_event=self.bootstrap_node.stop_wait_db_up_event)
+            abort_action()
+            LOGGER.info("Scylla was stopped succesfully on node %s", self.bootstrap_node.name)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.warning("Abort was failed on node %s with errof %s", self.bootstrap_node.name, exc)
+        finally:
+            self._set_wait_stop_event()
+
+    def clean_unbootstrapped_node(self):
+        node_host_ids = []
+        if found_stings := list(self.host_id_searcher):
+            for line in found_stings:
+                new_node_host_id = line.split(" ")[-1].strip()
+                LOGGER.debug("Node %s has host id: %s in log", self.bootstrap_node.name, new_node_host_id)
+                node_host_ids.append(new_node_host_id)
+        self.bootstrap_node.log.debug("New host was not properly bootstrapped. Terminate it")
+        self._terminate_node()
+        self.verification_node.raft.clean_group0_garbage(raise_exception=True)
+        if node_host_ids:
+            for host_id in node_host_ids:
+                self.verification_node.run_nodetool(
+                    f"removenode {host_id}", ignore_status=True, retry=3, warning_event_on_exception=True)
+
+        assert self.verification_node.raft.is_cluster_topology_consistent(), \
+            "Group0, Token Ring and number of node in cluster are differs. Check logs"
+        self.bootstrap_node.parent_cluster.check_nodes_up_and_normal()
+        LOGGER.info("Failed bootstrapped node %s removed. Cluster is in initial state", self.bootstrap_node.name)
+
+    @decorate_with_context(ignore_ycsb_connection_refused)
+    def _terminate_node(self):
```

Comment:
I'm not sure I like this spreading of the logic of terminating a node, even the one in nemesis.py is a bit too much, and now we have a duplication of it...

Reply:
Removed it

Reply:
remove it